test/extended/clusterup.sh

```diff
@@ -0,0 +1,194 @@
+function os::test::extended::clusterup::image () {
+    os::test::extended::clusterup::standard_test \
+        --image="registry.access.redhat.com/openshift3/ose" \
+        --version="v3.3"
+    os::cmd::expect_success_and_text "docker inspect origin" "registry.access.redhat.com/openshift3/ose:v3.3"
+    os::cmd::expect_success "oc cluster down"
+}
+
+# Tests creating a cluster with a public hostname
+function os::test::extended::clusterup::publichostname () {
+    os::test::extended::clusterup::standard_test \
+        --public-hostname="my.server.name"
+    os::cmd::expect_success "docker exec origin cat /var/lib/origin/openshift.local.config/master/master-config.yaml | grep 'masterPublicURL.*my\.server\.name'"
+    os::cmd::expect_success "oc cluster down"
+}
+
+# Tests creating a cluster with a numeric public hostname
+function os::test::extended::clusterup::numerichostname () {
+    os::test::extended::clusterup::standard_test \
+        --public-hostname="127.0.0.1"
+    os::cmd::expect_success "docker exec origin cat /var/lib/origin/openshift.local.config/master/master-config.yaml | grep 'masterPublicURL.*127\.0\.0\.1'"
+    os::cmd::expect_success "oc cluster down"
+}
+
+# Tests installation of metrics components
+function os::test::extended::clusterup::metrics () {
+    os::test::extended::clusterup::standard_test --metrics
+    os::test::extended::clusterup::verify_metrics
+    os::cmd::expect_success "oc cluster down"
+}
+
+# Tests installation of aggregated logging components
+function os::test::extended::clusterup::logging () {
+    os::test::extended::clusterup::standard_test --logging
+    os::test::extended::clusterup::verify_logging
+    os::cmd::expect_success "oc cluster down"
+}
+
+# Verifies that a service can be accessed by a peer pod
+# and by the pod running the service
+function os::test::extended::clusterup::svcaccess () {
+    os::cmd::expect_success "oc cluster up"
+    os::cmd::expect_success "oc create -f ${OS_ROOT}/examples/gitserver/gitserver-persistent.yaml"
+    os::cmd::try_until_success "oc get endpoints git | grep ':8080'" $(( 10*minute )) 1
+    # Test that a service can be accessed from a peer pod
+    os::cmd::expect_success "timeout 2m oc run peer --image=openshift/origin-gitserver:latest --attach --restart=Never --command -- curl http://git:8080/_/healthz"
+    # Test that a service can be accessed from the same pod
+    os::cmd::expect_success "timeout 2m oc rsh dc/git curl http://git:8080/_/healthz"
+    os::cmd::expect_success "oc cluster down"
+}
+
+# Verifies that cluster up can start when a process is bound
+# to the loopback interface with tcp port 53
+function os::test::extended::clusterup::portinuse () {
+    # Start listening on the host's 127.0.0.1 interface, port 53
+    docker run -d --name=port53 --entrypoint=/bin/bash --net=host openshift/origin-gitserver:latest -c "socat TCP-LISTEN:53,bind=127.0.0.1,fork SYSTEM:'echo hello'"
+    os::cmd::expect_success "oc cluster up"
```

Comment:
so this is a test for a very specific situation. When you're running dnsmasq, cluster-up fails to start. I don't think we need to make the test longer. If it starts successfully, it means we don't have the bug.